src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1426,6 +1426,40 @@ LogicalResult ONNXReshapeOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Resize
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXResizeOp::inferShapes() {
+  if (!X().getType().isa<RankedTensorType>())
+    return emitError("Input tensor not ranked");
+  auto xTy = X().getType().cast<RankedTensorType>();
+
+  if (!sizes().getType().isa<RankedTensorType>())
+    return emitError("Sizes tensor not ranked");
+  auto sizesTensorTy = sizes().getType().cast<RankedTensorType>();
+
+  // Only rank 1 sizes tensors are supported.
+  if (sizesTensorTy.getShape().size() != 1)
+    return emitError("Shape tensor must have rank one");
+
+  // Resize relies on shape information
+  // Failure if shape information cannot be retrieved
+  SmallVector<int64_t, 4> dims(xTy.getShape().size(), -1);
+  if (sizes().getDefiningOp()) {
```

Comment:
If there is no defining operation, I think we should error here instead of pushing in shape of -1's.

Reply:
Make sense, changing it!